producer/writer/message_writer.go

```diff
@@ -71,37 +67,40 @@ type messageWriter interface {
 
 	// SetCutoffNanos sets the cutoff nanoseconds.
 	SetCutoffNanos(nanos int64)
+
+	// QueueSize returns the number of messages queued in the writer.
+	QueueSize() int
 }
 
 type messageWriterMetrics struct {
 	writeSuccess           tally.Counter
 	oneConsumerWriteError  tally.Counter
 	allConsumersWriteError tally.Counter
+	noWritersError         tally.Counter
 	writeAfterCutoff       tally.Counter
 	writeBeforeCutover     tally.Counter
 	retryBatchLatency      tally.Timer
 	retryLatency           tally.Timer
-	messageQueueLength     tally.Counter
 }
 
 func newMessageWriterMetrics(scope tally.Scope) messageWriterMetrics {
 	return messageWriterMetrics{
-		writeSuccess: scope.Counter("write-success"),
-		oneConsumerWriteError: scope.
-			Tagged(map[string]string{"error-type": "one-consumer"}).
-			Counter("write-error"),
+		writeSuccess:          scope.Counter("write-success"),
+		oneConsumerWriteError: scope.Counter("write-error-one-consumer"),
```

Comment:
Why changing this one to use metric name instead of metric type to differentiate?

Reply:
Because this one is more like a warning, not as bad as other write errors